src/Service/Image.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace App\Service;
+
+use App\Exception\FilesystemException;
+use App\Exception\ImageFetchException;
+use Liip\ImagineBundle\Message\WarmupCache;
+use Symfony\Component\Filesystem\Exception\IOExceptionInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Filesystem\Path;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+final class Image implements ImageInterface
+{
+    public function __construct(
+        private readonly HttpClientInterface $client,
+        private readonly string $publicPath,
+        private readonly MessageBusInterface $messageBus,
+    ) {
+    }
+
+    /**
+     * @throws FilesystemException
+     * @throws TransportExceptionInterface
+     * @throws ImageFetchException
+     */
+    public function fetch(string $url): string
+    {
+        $filesystem = new Filesystem();
+
+        try {
+            $path = $this->generatePath(url: $url, absolute: true);
+            $filesystem->mkdir(
+                Path::normalize($path),
+                0775
+            );
+        } catch (IOExceptionInterface $exception) {
+            throw new FilesystemException('Unable to create upload folder', $exception->getCode(), $exception);
+        }
+
+        $response = $this->client->request('GET', $url);
+        if (200 !== $response->getStatusCode()) {
```

Comment:
```suggestion
        if (Response::HTTP_OK !== $response->getStatusCode()) {
```

https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpFoundation/Response.php#L28

Reply:
Fixed